### DISCUSSION TITLE
Statistics collection stuck under 50% complete but the bias correction success in 100%

### DISCUSSION THREAD

**Discussion by Zakihran13** [openvinotoolkit/nncf#1943] [Q&A]
Hai guys,
i just wondering how to fix this and whats really going on in my project. i have this YOLOv8 pre-trained model running in openVINO IR format that means i already use openVINO toolkit for the model inferencing. Here, i already run it perfectly and it works fine, but i wanted to try the model optimization that openVINO proposed to in YOLOv8 model tutorial.

![image](https://github.com/openvinotoolkit/nncf/assets/80832245/539e86f0-9b9c-445c-a549-a2cd5261536e)

i tried so hard to figure it out whats happen but i still dont know it (cause lack of knowledge i think). im using custom dataset and its working well with the the dataset (using dataloader). but the statistics collection is stuck like in the picture above. what is going on with this, is it okay or there's something wrong with the dataset or code that wrote?

thanks for helping me by the way...

**Comment by alexsu52** [comment]
@Zakihran13 thanks for the question. I can assume that your custom dataset contains only 40 samples and statistics collection is stopped when the end of the dataset is reached. By default, `nncf.quantize(...)` requires 300 samples to calculate activations statistics https://github.com/openvinotoolkit/nncf/blob/68b3037bc61c2cd1c5a1c6c22ab014b08d712d8a/nncf/quantization/quantize_model.py#L34, so the target of the statistic collection progress bar is 300 samples.
You can increase dataset size or feed `subset_size=40` to `nncf.quantize(...)`. But I absolutely agree with you that it is confused and we will try to fix it ASAP.

**Reply by Zakihran13** [reply]
ahhh i see....
thank you so much @alexsu52 for answering my question. i think my dataset for the validation is only 45 images. 
thank you so muchh....

**Reply by alexsu52** [reply]
PR: https://github.com/openvinotoolkit/nncf/pull/2114 with fix
